cdap-api/src/main/java/co/cask/cdap/api/dataset/DatasetSpecification.java

```diff
@@ -194,21 +208,25 @@ public DatasetSpecification setOriginalProperties(DatasetProperties originalProp
    * @return a new spec that is the same as this one, with the original properties set to the provided properties.
    */
   public DatasetSpecification setOriginalProperties(Map<String, String> originalProps) {
-    return new DatasetSpecification(name, type, originalProps, properties, datasetSpecs);
+    return new DatasetSpecification(name, type, description, originalProps, properties, datasetSpecs);
+  }
+
+  public DatasetSpecification setDescription(String description) {
```

Comment:
After discussion with @anew, we decided to add `setDescription()` to spec builder too. The description set in the spec will get overridden by any description given during instance creation.